parity-util-mem/src/malloc_size.rs

```diff
@@ -200,27 +200,61 @@ pub trait MallocConditionalShallowSizeOf {
     fn conditional_shallow_size_of(&self, ops: &mut MallocSizeOfOps) -> usize;
 }
 
-#[cfg(not(feature = "estimate-heapsize"))]
+#[cfg(not(any(
+	all(
+		target_os = "macos",
+		not(feature = "jemalloc-global"),
+	),
+	feature = "estimate-heapsize"
+)))]
+pub mod inner_allocator_use {
+
+use super::*;
+
+impl<T: ?Sized> MallocShallowSizeOf for Box<T> {
+    fn shallow_size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        unsafe { ops.malloc_size_of(&**self) }
+    }
+}
+
+impl<T> MallocShallowSizeOf for Vec<T> {
+    fn shallow_size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        unsafe { ops.malloc_size_of(self.as_ptr()) }
+    }
+}
+
+// currently this seems only fine with jemalloc
+#[cfg(feature = "std")]
+#[cfg(any(prefixed_jemalloc, target_os = "macos", target_os = "ios", target_os = "android", feature = "jemalloc-global"))]
```

Comment:
unrelated, but I think prefixed_jemalloc is coming [from heapsize](https://github.com/servo/heapsize/blob/32615cb931b4871b24f72114cca9faa8bca48399/build.rs#L34-L36) to support old very rust and we shouldn't use it.

Reply:
cc @NikVolf https://github.com/servo/heapsize/pull/44